querier.go

```diff
@@ -221,6 +221,36 @@ func (r *postingsReader) Select(ms ...labels.Matcher) (Postings, []string) {
 	return p, absent
 }
 
+// findTuplesByPrefix uses binary search to find prefix matches within ts.
+func findTuplesByPrefix(m *labels.PrefixMatcher, ts StringTuples) (res []string, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = r.(error)
+		}
+	}()
+	tslen := ts.Len()
+	i := sort.Search(tslen, func(i int) bool {
+		vs, err := ts.At(i)
+		if err != nil {
+			panic(fmt.Errorf("Failed to read tuple %d/%d: %v", i, tslen, err))
```

Comment:
So the problem is returning true won't interrupt the search process as the sort.Search will continue searching in the left subarray. Panicking is the only way you can interrupt the control flow in this case. // long rant on absence of exception in Go omitted

Anyway I don't feel strongly about it since it's only log(tslen) and supposed to be rare so lmkwyt.